### PR TITLE
Tab bar

### PR DESCRIPTION
**Pull request by patofernandez**
image

### CODE REVIEW

File: wbooks/src/app/components/TabBar/components/TabBarIcon/styles.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { StyleSheet } from 'react-native';
+
+import { lightBlue, lightGray } from '../../../../../constants/colors';
+
+export default StyleSheet.create({
+    container: {
+        alignItems: 'center'
+    },
+    image: { 
+        width: 21,
```
Comment (mattgle) [inline wbooks/src/app/components/TabBar/components/TabBarIcon/styles.js]: extract 21 and use it here and in the next line (11)

File: wbooks/src/app/screens/BookList/index.js [hunk 2]
```diff
@@ -33,4 +35,4 @@ class BookList extends Component {
 	}
 }
 
-export default BookList;
+export default BookList
```
Comment (mattgle) [inline wbooks/src/app/screens/BookList/index.js]: add `;`

Comment (patofernandez) [reply wbooks/src/app/screens/BookList/index.js]: done

File: wbooks/src/app/components/TabBar/components/TabBarIcon/styles.js (hunk 1 above)
Comment (patofernandez) [reply wbooks/src/app/components/TabBar/components/TabBarIcon/styles.js]: done!

File: wbooks/App.js [hunk 3]
```diff
@@ -1,28 +1,54 @@
...
+          tabBarLabel: null,
+          tabBarIcon: ({focused}) => <TabBarIcon activeIcon={SettingsActiveIcon} inactiveIcon={SettingsIcon} label={'Settings'} focused={focused}/>
+        }
+      },
+    },
+    {
+      tabBarOptions: {
+        showLabel: false,
+        activeTintColor: 'red',
+        inactiveTintColor: 'gray',
```
[the remark this answers is not in the document]
Comment (patofernandez) [reply wbooks/App.js]: done

File: wbooks/App.js [hunk 4]
```diff
@@ -1,28 +1,54 @@
...
+        navigationOptions: {
+          tabBarLabel: null,
+          tabBarIcon: ({focused}) => <TabBarIcon activeIcon={SettingsActiveIcon} inactiveIcon={SettingsIcon} label={'Settings'} focused={focused}/>
+        }
+      },
+    },
+    {
+      tabBarOptions: {
+        showLabel: false,
+        activeTintColor: 'red',
```
[the remark this answers is not in the document]
Comment (patofernandez) [reply wbooks/App.js]: done

File: wbooks/App.js [hunk 5]
```diff
@@ -1,28 +1,54 @@
...
+import SettingsIcon from './src/assets/ic_settings.png';
 import BookList from './src/app/screens/BookList';
 import BookDetail from './src/app/screens/BookDetail';
 import CustomHeader from './src/app/components/Header';
+import TabBarIcon from './src/app/components/TabBar/components/TabBarIcon';
+
+class SettingsScreen extends Component {
+  render() {
+    return (
+      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
```
[the remark this answers is not in the document]
Comment (patofernandez) [reply wbooks/App.js]: done

File: wbooks/App.js [hunk 6]
```diff
@@ -1,28 +1,54 @@
...
+import BookListActiveIcon from './src/assets/ic_library_active.png';
+import BookListIcon from './src/assets/ic_library.png';
+import SettingsActiveIcon from './src/assets/ic_settings_active.png';
+import SettingsIcon from './src/assets/ic_settings.png';
 import BookList from './src/app/screens/BookList';
 import BookDetail from './src/app/screens/BookDetail';
 import CustomHeader from './src/app/components/Header';
+import TabBarIcon from './src/app/components/TabBar/components/TabBarIcon';
+
+class SettingsScreen extends Component {
```
[the remark this answers is not in the document]
Comment (patofernandez) [reply wbooks/App.js]: done

File: wbooks/App.js [hunk 7]
```diff
@@ -1,28 +1,43 @@
...
+          tabBarLabel: null,
+          tabBarIcon: ({focused}) => <TabBarIcon activeIcon={SettingsActiveIcon} inactiveIcon={SettingsIcon} label={'Settings'} focused={focused}/>
+        }
+      },
+    },
+    {
+      tabBarOptions: {
+        showLabel: false
+      },
+    }
```
Comment (guidoprinc) [inline wbooks/App.js]: extract all navigationOptions and tabBarOptions in a file located in the folder `config`. The file must be called `navigationOptions.js`

File: wbooks/src/app/components/TabBar/components/TabBarIcon/index.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { View, Image, Text } from 'react-native';
+
+import styles from './styles';
+
+const TabBarIcon = ({ activeIcon, inactiveIcon, focused, label }) => (
+  <View style={styles.container}>
+    <Image style={styles.image} source={focused ? activeIcon : inactiveIcon } resizeMode="cover"/>
```
Comment (guidoprinc) [inline wbooks/src/app/components/TabBar/components/TabBarIcon/index.js]: delete right space on source, after inactiveIcon

File: wbooks/src/app/components/TabBar/components/TabBarIcon/index.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    <Text style={[styles.label, focused && styles.activeLabel]}>{label}</Text>
+  </View>
+);
+
```
Comment (guidoprinc) [inline wbooks/src/app/components/TabBar/components/TabBarIcon/index.js]: you are missing the propTypes

File: wbooks/src/app/screens/Settings/styles.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { StyleSheet } from 'react-native';
+
+export default StyleSheet.create({
+  container: { 
+    flex: 1, 
+    justifyContent: 'center'
+    , alignItems: 'center' 
+  }
+});
```
Comment (guidoprinc) [inline wbooks/src/app/screens/Settings/styles.js]: EOF missing

File: wbooks/src/app/components/TabBar/components/TabBarIcon/index.js (hunk 8 above)
Comment (patofernandez) [reply wbooks/src/app/components/TabBar/components/TabBarIcon/index.js]: done

File: wbooks/src/app/screens/Settings/styles.js (hunk 10 above)
Comment (patofernandez) [reply wbooks/src/app/screens/Settings/styles.js]: done

File: wbooks/App.js (hunk 7 above)
Comment (patofernandez) [reply wbooks/App.js]: done

File: wbooks/src/app/components/TabBar/components/TabBarIcon/index.js (hunk 9 above)
Comment (patofernandez) [reply wbooks/src/app/components/TabBar/components/TabBarIcon/index.js]: porpTypes added

File: wbooks/src/config/navigationOptions.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+export const settingsNavigationOptions = {
+  title: 'Settings',
+  tabBarLabel: null,
+  tabBarIcon: ({focused}) => <TabBarIcon activeIcon={SettingsActiveIcon} inactiveIcon={SettingsIcon} label={'Settings'} focused={focused}/>
+};
+
+export const tabBarOptions = {
+  showLabel: false
+};
```
Comment (guidoprinc) [inline wbooks/src/config/navigationOptions.js]: EOF

File: wbooks/src/app/screens/Settings/index.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import React, { Component } from 'react';
+import { View, Text } from 'react-native';
+import styles from './styles';
+
+const SettingsScreen = () => (
```
Comment (mattgle) [inline wbooks/src/app/screens/Settings/index.js]: Change this arrow function for a function.

File: wbooks/src/config/navigationOptions.js (hunk 11 above)
Comment (patofernandez) [reply wbooks/src/config/navigationOptions.js]: done

File: wbooks/src/app/screens/Settings/index.js (hunk 12 above)
Comment (patofernandez) [reply wbooks/src/app/screens/Settings/index.js]: done

**Pull request merged**